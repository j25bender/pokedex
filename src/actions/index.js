export const addTypes = (typesToDispatch) => ({
   type: 'ADD_TYPES',
   typesToDispatch
})

export const addPokemon = (pokemonToDispatch) => ({
  type: 'ADD_POKEMON',
  pokemonToDispatch
})


