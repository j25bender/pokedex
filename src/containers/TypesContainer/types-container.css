.types-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  background-color: #fefaf3;  
}

.card,
.active {
  border: 2px black solid;
  margin: 10px;
  height: 50px;
  width: 300px;
  overflow-y: scroll;
}

div.card {
  font-size: 20px;
}

.active {
  border: 2px black solid;
  margin: 10px;
  height: 300px;
  width: 300px;  
  transition: height .25s ease;
  font-size: 25px;
  font-weight: bold;
}

.active h6 {
  font-size: 16px;
}

.pokeCategory {
  max-height: 275px;
}

.loadScreen {
  position: absolute;
  width: 100%;
  height: 100%;
  background: url('https://secure.static.tumblr.com/7c9fb5875d110c5c7ef59a9e9432c753/rjjxcxi/ENsnjqzs7/tumblr_static_tumblr_static_qpbmgd6m7c0gcw8cg4ogg040_640.gif') no-repeat center center;
}