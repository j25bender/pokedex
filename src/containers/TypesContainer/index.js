import React, { Component } from 'react';
import './types-container.css';
import PropTypes, { shape, func, string } from 'prop-types';
import { connect } from 'react-redux';
import { addTypes, addPokemon } from '../../actions';

export class TypesContainer extends Component {
  constructor() {
    super()

    this.state = {
      types: [],
      pokemonCategory: []
    }
  }

  handleClick(e) {
    const { storePokemon } = this.props
    e.target.className === "card" ? e.target.className = "active" : e.target.className = "card";
    const pokemonCategory = storePokemon.filter( pokemon => pokemon.type === e.target.id ? pokemon : null )
    this.setState({ pokemonCategory });
  }

  async componentDidMount() {
    const fetchedTypes = await fetch('http://localhost:3001/types');
    const typesToDispatch = await fetchedTypes.json();
    const fetchedPokemon = await fetch('http://localhost:3001/pokemon');
    const pokemonToDispatch = await fetchedPokemon.json();
    this.props.addPokemon(pokemonToDispatch);
    this.props.addTypes(typesToDispatch);
    this.setState({ 
      types: typesToDispatch,
      pokemon: pokemonToDispatch
     })
  }
  
  render(props) {
    const { storeTypes } = this.props
    const { pokemonCategory } = this.state
    console.log(pokemonCategory)
    const renderPokemon = pokemonCategory.map( (poke, key) => <div className="pokeCategory" key={ poke.id } ><h6>{ poke.name }</h6><h6>{ poke.type }</h6><h6>{ poke.weight }</h6><img src={ poke.sprites.front_default} /></div>)

    const renderCards = storeTypes.map( (type, key) => <div id={ type.id } className="card" onClick={ (e) => this.handleClick(e) } key={ type.id } >{ type.name }<div>{ renderPokemon }</div></div>)
    
    const loading = !this.props ? <div className="loadScreen"></div> : null;

    return (
      <div className="types-container">
        {loading} 
        {renderCards}
      </div>
    );
  }
}

TypesContainer.propTypes = {
  addTypes: PropTypes.func
};

export const mapStateToProps = (state) => ({
   storeTypes: state.storeTypes,
   storePokemon: state.storePokemon 
});

export const mapDispatchToProps = (dispatch) => ({
  addTypes: (typesToDispatch) => dispatch(addTypes(typesToDispatch)),
  addPokemon: (pokemonToDispatch) => dispatch(addPokemon(pokemonToDispatch))
});

export default connect(mapStateToProps, mapDispatchToProps)(TypesContainer);
