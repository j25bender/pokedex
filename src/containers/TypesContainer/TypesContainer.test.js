import React, { Component } from 'react';
import { connect } from 'react-redux';
import { addTypes, addPokemon } from '../../actions';
import TypesContainer from './index';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux'
import { createStore } from 'redux'
import configureStore from 'redux-mock-store';

describe('TypesContainer Tests', () => {
  const typesReducerMock = [{id: "1", name: "normal", pokemon: Array(5)}, {id: "2", name: "fighting", pokemon: Array(5)}]
  const pokemonReducerMock = [ {name: "gastly", id: "92", type: "8", weight: 1, sprites: 'img'}, {name: "haunter", id: "93", type: "8", weight: 1, sprites: 'img'} ]
  const mockHandleClick = jest.fn();
  
  const initialState = {
    storeTypes: typesReducerMock,
    storePokemon: pokemonReducerMock
  }
  const mockStore = configureStore()
  let store, wrapper 

  beforeEach(() => {
    store = mockStore(initialState)
    wrapper = shallow(<TypesContainer store={ store }/>)
  })

  it('should match snapshot', () => {
    expect(wrapper.length).toEqual(1);
    expect(wrapper).toMatchSnapshot();
  })
})


// const store = createStore(mockStore)

// window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
//   json: () => Promise.resolve({
//     typesToDispatch: mocktypesToDispatch
//   })
// }))

// let wrapper