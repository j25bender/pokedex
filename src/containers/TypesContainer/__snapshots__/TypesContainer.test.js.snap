// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TypesContainer Tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Connect(TypesContainer)
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "addPokemon": [Function],
      "addTypes": [Function],
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "storePokemon": Array [
        Object {
          "id": "92",
          "name": "gastly",
          "sprites": "img",
          "type": "8",
          "weight": 1,
        },
        Object {
          "id": "93",
          "name": "haunter",
          "sprites": "img",
          "type": "8",
          "weight": 1,
        },
      ],
      "storeSubscription": Subscription {
        "listeners": Object {
          "clear": [Function],
          "get": [Function],
          "notify": [Function],
          "subscribe": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "unsubscribe": [Function],
      },
      "storeTypes": Array [
        Object {
          "id": "1",
          "name": "normal",
          "pokemon": Array [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ],
        },
        Object {
          "id": "2",
          "name": "fighting",
          "pokemon": Array [
            undefined,
            undefined,
            undefined,
            undefined,
            undefined,
          ],
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "addPokemon": [Function],
        "addTypes": [Function],
        "store": Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        },
        "storePokemon": Array [
          Object {
            "id": "92",
            "name": "gastly",
            "sprites": "img",
            "type": "8",
            "weight": 1,
          },
          Object {
            "id": "93",
            "name": "haunter",
            "sprites": "img",
            "type": "8",
            "weight": 1,
          },
        ],
        "storeSubscription": Subscription {
          "listeners": Object {
            "clear": [Function],
            "get": [Function],
            "notify": [Function],
            "subscribe": [Function],
          },
          "onStateChange": [Function],
          "parentSub": undefined,
          "store": Object {
            "clearActions": [Function],
            "dispatch": [Function],
            "getActions": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
          },
          "unsubscribe": [Function],
        },
        "storeTypes": Array [
          Object {
            "id": "1",
            "name": "normal",
            "pokemon": Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          },
          Object {
            "id": "2",
            "name": "fighting",
            "pokemon": Array [
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
            ],
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
